### PR TITLE
feat(console): error handling in dashboard

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Handle dashboard errors in general: if any API request occurs error, stop rendering page, show a page-level error box.

## Linear Issue Reference

LOG-2808

## Testing

截屏2022-06-09 下午4 23 21

### CODE REVIEW

File: packages/console/src/pages/Dashboard/components/Error/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const Error = ({ errorMessage }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  return (
+    <div className={styles.error}>
+      <img src={ErrorImage} />
+      <div className={styles.title}>{errorMessage ?? t('errors.something_went_wrong')}</div>
+    </div>
+  );
+};
```
Comment (demonzoo) [inline packages/console/src/pages/Dashboard/components/Error/index.tsx:21]: Can we just re-use `<AppError>` component?

File: packages/console/src/pages/Dashboard/index.tsx [hunk 2]
```diff
@@ -33,7 +43,8 @@ const Dashboard = () => {
         <div className={styles.subtitle}>{t('dashboard.description')}</div>
       </div>
       {isLoading && <Skeleton />}
-      {!isLoading && (
+      {!isLoading && error && <Error errorMessage={error.body?.message} />}
+      {!isLoading && totalData && newData && activeData && (
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/console/src/pages/Dashboard/index.tsx:47]: This is aligned with @fleuraly, and they are very likely to have the same status(operational or crashdown), display them togother when all of them are ready is a good experience.

File: packages/console/src/pages/Dashboard/index.tsx [hunk 3]
```diff
@@ -33,7 +43,8 @@ const Dashboard = () => {
         <div className={styles.subtitle}>{t('dashboard.description')}</div>
       </div>
       {isLoading && <Skeleton />}
-      {!isLoading && (
+      {!isLoading && error && <Error errorMessage={error.body?.message} />}
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/console/src/pages/Dashboard/index.tsx]: updated

File: packages/console/src/pages/Dashboard/components/Error/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/Dashboard/components/Error/index.tsx]: updated

**Review by demonzoo — APPROVED** [review]

**Comment by wangsijie** [comment]
image

**Pull request merged**